cocos/core/renderer/scene/spot-light.ts

```diff
@@ -118,14 +121,42 @@ export class SpotLight extends Light {
     }
 
     set luminance (lum: number) {
+        const isHDR = (legacyCC.director.root).pipeline.pipelineSceneData.isHDR;
+        if (isHDR) {
+            this._luminance = lum;
+        } else {
+            this._luminance_ldr = lum;
+        }
+
+        if (JSB) {
+            (this._nativeObj! as NativeSpotLight).setIlluminance(this._luminance);
+            (this._nativeObj! as NativeSpotLight).setIlluminance_ldr(this._luminance_ldr);
+        }
+    }
+
+    get luminance (): number {
+        const isHDR = (legacyCC.director.root).pipeline.pipelineSceneData.isHDR;
+        if (isHDR) {
+            return this._luminance;
+        } else {
+            return this._luminance_ldr;
+        }
+    }
+
+    set luminance_hdr (lum: number) {
```

Comment:
Naming:

- DirectionalLight
    - illuminance
    - illuminance_hdr
    - illuminance_ldr
- SpotLight
    - luminance
    - luminance_ldr
- SphereLight
    - luminance
    - luminance_hdr
    - luminance_ldr

While in JSB
- DirectionalLight: setIlluminance
- PointLight: setIlluminance
- SphereLight: setIlluminance

Please unify the API naming rule, and make sure deprecation and data migration